### PR TITLE
Add multiple option for args

### PR DESCRIPTION
**Pull request by nat-n**
WIP solution for #54 @wsascha

The new multiple option can be added to args to make them accept multiple values.

- only the last positional arg may have multiple set
- multiple option defaults to false
- if `multiple = true` and `required = false` then the result is `nargs = "*"`, or if `required = true` then `nargs = "+"`
- an integer value >= 2 may also be given to specify an exact number of values are required for an arg
- for script tasks the values are passed as a list, for command tasks they can be templated into the command (thus potentially passed as multiple arguments), shell tasks don't have any special support.
- All tasks also get the list of values as an env var in the form of a space delimited string

## Still to test:
- [x] created a task for testing in the scripts project, but still need to write thorough test cases yet in `test_script_tasks`
- [x] also need basic test cases for cmd, shell, and sequence tasks
- [x] what if multiple=true with type=boolean ?

## Also todo:
- [x] Fix other broken tests
- [x] Document new functionality in the readme
- [ ] Think of a nice way to indicate that an arg accepts multiple arguments when generating documentation for a task

### CODE REVIEW

File: poethepoet/task/args.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "positional": (bool, str),
     "required": bool,
     "type": str,
+    "multiple": (bool, int),
```
Comment (wsascha) [inline poethepoet/task/args.py:29]: Not sure if it's really necessary to have an int here. At least I don't know any use cases from tools I used.

File: tests/fixtures/scripts_project/pyproject.toml [hunk 2]
```diff
@@ -99,6 +99,29 @@ greet = "pkg:greet"
     type = "boolean"
 
 
+  [tool.poe.tasks.multiple-value-args]
```
Comment (wsascha) [inline tests/fixtures/scripts_project/pyproject.toml:102]: Maybe we could also add a sequence task here.

**Review by wsascha — COMMENTED** [review]
Hey, sorry for the late reply, I'm currently on vacation 😅  looks good to me over all. Thanks for your efforts, really appreciated! I left two minor comments below, let me know what you think.

File: poethepoet/task/args.py (hunk 1 above)
Comment (nat-n) [reply poethepoet/task/args.py:29]: I'm not sure either. Though it seemed like a low cost feature to support, and imagined that there could be use-cases for  it.

**Pull request merged**